scripts/build-controller-release.sh

```diff
@@ -36,6 +36,7 @@ ACK_GENERATE_BIN_PATH=${ACK_GENERATE_BIN_PATH:-$DEFAULT_ACK_GENERATE_BIN_PATH}
 ACK_GENERATE_API_VERSION=${ACK_GENERATE_API_VERSION:-"v1alpha1"}
 ACK_GENERATE_CONFIG_PATH=${ACK_GENERATE_CONFIG_PATH:-""}
 ACK_GENERATE_IMAGE_REPOSITORY=${ACK_GENERATE_IMAGE_REPOSITORY:-"$DEFAULT_IMAGE_REPOSITORY"}
+AWS_SDK_GO_VERSION=${AWS_SDK_GO_VERSION:-"v1.35.5"}
```

Comment:
Latest is `1.38.33`. Should we move the needle or would this introduce bugs?

Reply:
Good point. +1 for using the latest - worst case we can set the older version..

Reply:
I'll leave the version as `v1.35.5` for now, it's better to do it separately since other files also use 1.35.5 as default version

Reply:
Do we know how these are coupled? Is it a good idea to make it into an ENV variable if we aren't comfortable changing it?

Reply:
The only scripts using `AWS_SDK_GO_VERSION` variable are `build-controller.sh`  and `build-controller-release.sh`, and AFAIK they not really coupled, both scripts call `ack-generate` for different purposes. The only scenario that i'm concerned about (if the defaults values are not synced) is if a user calls both scripts without setting `AWS_SDK_GO_VERSION`. 

> Is it a good idea to make it into an ENV variable if we aren't comfortable changing it?

IMO It's not super dangerous, it's just safer to keep them in sync. Is there a different approach if we don't make it into an ENV variable?

Reply:
> both scripts call `ack-generate` for different purposes

I suppose this is the confusing part to me. Is this something we should address with a clearer configuration? 

Nevertheless I'm happy to merge this PR for now to unblock @komish . But I think I'll bring this up in our next meeting :)

Reply:
@RedbackThomson For what it's worth - at this point it isn't a hard block because we're not actively publishing/releasing OpenShift/OLM related components. I'll be able to launch my PRs soon without this being merged with the understanding that when the Go SDK version discrepancy gets sorted, things will fall into place. 👍

Reply:
> Is this something we should address with a clearer configuration?

Definitely! I think this is an area we still can improve ++